git-flow/rebase.md

```diff
@@ -0,0 +1,111 @@
+
+## What is Rebase?
+
+Rebase is a Git command that allows you to move or combine a sequence of commits to a new base commit. It re-writes the commit history by replaying each commit on top of a different commit or branch. This can lead to a cleaner, linear history, reducing the number of merge commits.
+
+## When to Use Rebase?
+
+Rebase is particularly useful in the following scenarios:
+
+- Keeping a clean commit history: By rebasing, you can create a linear history without unnecessary merge commits, making it easier to track changes and understand the project's evolution.
+- Preparing for pull requests: Before submitting a pull request, you can rebase your branch onto the latest changes from the target branch to avoid potential conflicts.
+- Squashing commits: Rebase allows you to combine multiple commits into one, keeping the commit history clean and concise.
+
+## How to Rebase
+
+### Interactive Rebase
+
+To perform an interactive rebase, follow these steps:
+
+1. **Open the terminal**: Navigate to your repository using the command line.
+
+2. **Fetch the latest changes**: Run `git fetch` to get the latest updates from the remote repository.
+
+3. **Choose the base commit**: Identify the commit you want to use as the new base. You can use a branch name or a commit hash.
+
+4. **Initiate rebase**: Execute the command `git rebase -i <base-commit>`. Or you can use the another way `git rebase -i HEAD~<number-of-commits>`. For example, if you want to rebase the last 3 commits, you can use `git rebase -i HEAD~3`.
+
+5. **Interactive editor**: An interactive editor will open, showing a list of commits that will be included in the rebase. You can choose to re-order, squash, edit, or drop commits. Here the example how it look like 
+```
+pick 68ed932 [FEC-53]-validation-yup
+pick 58824a6 [FEC-53]-add-react-hook-form
+pick 2704e35 [FEC-53]-update-hook-form
+pick 248c188 [FEC-53]-add-formik
+pick 07e292b [FEC-53]-update-formik
+
+# Rebase 46ed046..ccb84f0 onto 46ed046 (5 commands)
+#
+# Commands:
+# p, pick <commit> = use commit
+# r, reword <commit> = use commit, but edit the commit message
+# e, edit <commit> = use commit, but stop for amending
+# s, squash <commit> = use commit, but meld into previous commit
+# f, fixup [-C | -c] <commit> = like "squash" but keep only the previous
+#                    commit's log message, unless -C is used, in which case
+#                    keep only this commit's message; -c is same as -C but
+#                    opens the editor
+# x, exec <command> = run command (the rest of the line) using shell
+"~/WorkSpace/fe-cookbook/.git/rebase-merge/git-rebase-todo" 33L, 1465B
+```
+6. **Save and exit**: After making the desired changes, save the file and exit the editor. In command line, you can use `:wq` to save and exit.
```

Comment:
I also would  add how to edit the interactive editor 
 (if you want to squash or something else)